istioctl/pkg/tag/generate.go

```diff
@@ -149,6 +151,29 @@ func Generate(ctx context.Context, client kube.Client, opts *GenerateOptions, is
 	return tagWhYAML, nil
 }
 
+func fixWhFailurePolicy(client kube.Client, vwc *admitv1.ValidatingWebhookConfiguration) (*admitv1.ValidatingWebhookConfiguration, error) {
+	var curWebhooks []admitv1.ValidatingWebhook
+	curObj, err := client.Kube().AdmissionregistrationV1().ValidatingWebhookConfigurations().
+		Get(context.Background(), vwc.Name, metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return nil, err
+		}
+	}
+	if curObj != nil {
+		curWebhooks = curObj.Webhooks
+	}
+	for i := range vwc.Webhooks {
+		if curWebhooks != nil && i < len(curWebhooks) {
+			failurePolicy := curWebhooks[i].FailurePolicy
```

Comment:
this seems suspicious. The indexes are probably the same but could not be. I think my comment about re-using the existing code probably makes it easier